libs/analytics/src/types.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Shared analytics types
+ *
+ * This is the single source of truth for all analytics categories across the CoW Protocol applications.
+ * When adding new categories, please add them here and use them through imports.
+ *
+ * Example usage with data attributes:
+ * ```tsx
+ * import { Category, toGtmEvent } from '@cowprotocol/analytics'
+ *
+ * function MyComponent() {
+ *   return (
+ *     <button
+ *       data-click-event={toGtmEvent({
+ *         category: Category.TRADE,
+ *         action: 'Confirm Trade',
+ *         label: 'Swap'
+ *       })}
+ *     >
+ *       Confirm
+ *     </button>
+ *   )
+ * }
+ * ```
+ */
+
+export enum Category {
```

Comment:
This is an inverted dependency.
Since this is a library and it supposed to be generic, it should not be aware of specific applications.
I would split this enum and put the values in specific app code.

Reply:
Addressed this now (quite the overhaul..)

Reply:
Great job! Thanks!